File: py/leetcode-fizzbuzz.py
class Solution:
    def fizzBuzz(self, n: int) -> list[str]:
        if n == 1:
            return ["1"]
        l = Solution.fizzBuzz(self, n - 1)
        if n % 3 ==0 and n % 5 == 0:
            l.append("FizzBuzz")
        elif n % 3 == 0:
            l.append("Fizz")
        elif n % 5 == 0:
            l.append("Buzz")
        else:
            l.append(str(n))
        return l

s = Solution()
print(s.fizzBuzz(15))

File: java/leetcode-fizzbuzz.java
import java.util.ArrayList;

class Solution {
    public List<String> fizzBuzz(int n) {
        ArrayList<String> r = new ArrayList();
        if (n == 1) {
            r.add("1");
            return r;
        }
        List<String> prev = fizzBuzz(n - 1);
        if (n % 3 == 0 && n % 5 == 0) {
            prev.add("FizzBuzz");
        }
        else if (n % 3 == 0) {
            prev.add("Fizz");
        }
        else if (n % 5 == 0) {
            prev.add("Buzz");
        }
        else {
            prev.add("" + n);
        }
        return prev;
    }
}